sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeExtractors.scala

```diff
@@ -86,4 +86,57 @@ case class GpuGetArrayItem(child: Expression, ordinal: Expression)
       }
     }
   }
+}
+
+class GpuGetMapValueMeta(
+  expr: GetMapValue,
+  conf: RapidsConf,
+  parent: Option[RapidsMeta[_, _, _]],
+  rule: ConfKeysAndIncompat)
+  extends BinaryExprMeta[GetMapValue](expr, conf, parent, rule) {
+  import GpuOverrides._
+
+  override def tagExprForGpu(): Unit = {
+    if (!isStringLit(expr.key)) {
+      willNotWorkOnGpu("Only String literal keys are supported")
+    }
+  }
+
+  override def convertToGpu(
+    child: Expression,
+    key: Expression): GpuExpression =
+    GpuGetMapValue(child, key)
+
+  def isSupported(t: DataType) = t match {
+    case MapType(StringType, StringType, true) => true
+    case StringType => true
+  }
+
+  override def areAllSupportedTypes(types: DataType*): Boolean = types.forall(isSupported)
+}
+
+case class GpuGetMapValue(child: Expression, key: Expression)
+  extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+
+  override def dataType: DataType = child.dataType.asInstanceOf[MapType].valueType
+  override def inputTypes: Seq[AbstractDataType] =
```

Comment:
The original GetMapValue has a `checkInputDataTypes` that verifies that the key types match. Do we want/need to support it too?

Reply:
I didnt think it was really necessary but I should check it out again!

Reply:
Added it in as it is in Spark. I am not exactly sure how to make sure the ordering piece works there so did not add that here.